tests/cli.rs

```diff
@@ -0,0 +1,225 @@
+#![cfg(any(target_os = "windows", target_os = "macos", target_os = "linux"))]
+use std::env;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::{Path, PathBuf};
+use std::process::{Child, ChildStderr, Command, Stdio};
+
+use anyhow::{Context, Result};
+use tempfile::tempdir;
+
+#[test]
+fn cli_provide_one_file() -> Result<()> {
+    // provide a path to a file, do not pipe from stdin, do not pipe to stdout
+    test_provide_get_loop(&PathBuf::from("transfer").join("foo.bin"), false, false)
+}
+
+#[test]
+fn cli_provide_folder() -> Result<()> {
+    // provide a path to a folder, do not pipe from stdin, do not pipe to stdout
+    test_provide_get_loop(&PathBuf::from("transfer"), false, false)
+}
+
+#[test]
+fn cli_provide_from_stdin_to_stdout() -> Result<()> {
+    // provide a file, pipe content to the provider's stdin, pipe content to the getter's stdout
+    test_provide_get_loop(&PathBuf::from("transfer").join("foo.bin"), true, true)
+}
+
+// Test the provide and get loop for success, stderr output, and file contents.
```

Comment:
Not that this matters much, but might as well use doc comments.  Often can be nice to look at `cargo doc --document-private-items`

Reply:
🤦‍♀️ i meant to do doc comments... ty